cmd/zoekt-webserver/main.go

```diff
@@ -424,11 +437,72 @@ func (s *loggedSearcher) StreamSearch(ctx context.Context, q query.Q, opts *zoek
 		mu.Unlock()
 		sender.Send(event)
 	}))
+
+	s.log(ctx, q, opts, &stats, err)
+
+	return err
+}
+
+func (s *loggedSearcher) log(ctx context.Context, q query.Q, opts *zoekt.SearchOptions, st *zoekt.Stats, err error) {
+	id := traceID(ctx)
 	if err != nil {
-		log.Printf("EROR: search failed q=%s: %s", q.String(), err.Error())
+		log.Printf("EROR: search failed traceID=%s q=%s: %s", id, q.String(), err.Error())
+		return
 	}
-	log.Printf("DBUG: search q=%s Options{EstimateDocCount=%v Whole=%v ShardMaxMatchCount=%v TotalMaxMatchCount=%v ShardMaxImportantMatch=%v TotalMaxImportantMatch=%v MaxWallTime=%v MaxDocDisplayCount=%v} Stats{ContentBytesLoaded=%v IndexBytesLoaded=%v Crashes=%v Duration=%v FileCount=%v ShardFilesConsidered=%v FilesConsidered=%v FilesLoaded=%v FilesSkipped=%v ShardsSkipped=%v MatchCount=%v NgramMatches=%v Wait=%v}", q.String(), opts.EstimateDocCount, opts.Whole, opts.ShardMaxMatchCount, opts.TotalMaxMatchCount, opts.ShardMaxImportantMatch, opts.TotalMaxImportantMatch, opts.MaxWallTime, opts.MaxDocDisplayCount, stats.ContentBytesLoaded, stats.IndexBytesLoaded, stats.Crashes, stats.Duration, stats.FileCount, stats.ShardFilesConsidered, stats.FilesConsidered, stats.FilesLoaded, stats.FilesSkipped, stats.ShardsSkipped, stats.MatchCount, stats.NgramMatches, stats.Wait)
-	return err
+
+	if st == nil {
+		return
+	}
+
+	minDuration, _ := time.ParseDuration(os.Getenv("SRC_HTTP_LOG_MIN_DURATION"))
```

Comment:
I find having every query logged quite useful for zoekt. It isn't as noisy as other logs. Can we either not include this or change the default to minDuration = 0

Also probably over thinking it, but would be great to just parse this once? It's essentially a global so might as well parse as a global? Or parse in the duration as a field to loggedSearcher. Or finally, just remove this feature :)

Reply:
> Or finally, just remove this feature :)

Will go for this one if you don't think logging every query is a problem :)